### PR TITLE
Avoid possible NPE in Writer close

### PR DESCRIPTION
**Pull request by felixYyu**
cc: @rdblue @RussellSpitzer

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/ManifestWriter.java [hunk 1]
```diff
@@ -178,8 +178,10 @@ public ManifestFile toManifestFile() {
 
   @Override
   public void close() throws IOException {
-    this.closed = true;
-    writer.close();
+    if (writer != null) {
+      writer.close();
+      this.closed = true;
```
Comment (kbendick) [inline core/src/main/java/org/apache/iceberg/ManifestWriter.java:183]: Nit: This should probably still set `this.closed = true` even if the writer is null.

**Review by kbendick — COMMENTED** [review]
Is it possible for any of these writers to actually be `null`? It seems like they're all created in their respective constructors but maybe you've encountered something I haven't.

**Comment by felixYyu** [comment]
Refer to other writing methods, Possible NPE risks, I think we should try to avoid it.
e.g. AvroFileAppender
```
@Override
  public void close() throws IOException {
    if (writer != null) {
      writer.close();
      this.writer = null;
      isClosed = true;
    }
  }
```

**Review by hililiwei — COMMENTED** [review]
As @kbendick says, it's unlikely that it will be Null (at least I haven't encountered it yet), but for serious reasons, I'm in favor of this PR.

**Comment by rdblue** [comment]
These aren't going to be null because they're set in the constructor and don't change. I don't see much value in changes like this.

**Comment by felixYyu** [comment]
@rdblue 

> These aren't going to be null because they're set in the constructor and don't change. I don't see much value in changes like this.

OK, I'll close this PR.

**Pull request closed without merging**